### PR TITLE
nixos/matrix-synapse: minor improvements to implement worker-support

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

This doesn't contain changes for actual worker-support as I haven't figured out (yet) how to properly implement this in this module.

However, there are two changes that I needed to implement it in my personal Matrix configuration:

* Expose config file for matrix-synapse as read-only option to make sure that custom worker services can access it.
* Expose `synapse.app.generic_worker` as executable in `pkgs.matrix-synapse`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by sumnerevans — CHANGES_REQUESTED** [review]

**Review by sumnerevans — APPROVED** [review]

**Review by sumnerevans — CHANGES_REQUESTED** [review]
Actually, we should also set `enableRedis` to `true` by default in the package.

**Comment by dali99** [comment]
> Actually, we should also set `enableRedis` to `true` by default in the package.

We do this since #140207 :smile:

**Review by fadenb — APPROVED** [review]

**Review by sumnerevans — APPROVED** [review]

File: pkgs/servers/matrix-synapse/0002-Expose-generic-worker-as-binary-under-NixOS.patch [hunk 1]
```diff
@@ -0,0 +1,43 @@
+From 3089758015c64cc1e6788793c4fe40a0e1783457 Mon Sep 17 00:00:00 2001
+From: Maximilian Bosch <maximilian@mbosch.me>
+Date: Tue, 5 Oct 2021 22:33:12 +0200
+Subject: [PATCH 2/2] Expose generic worker as binary under NixOS
```
Comment (andir) [inline pkgs/servers/matrix-synapse/0002-Expose-generic-worker-as-binary-under-NixOS.patch:4]: Is there a reason we haven't upstreamed this? That looks like something that wouldn't hurt upstream at all.

Comment (Ma27) [reply pkgs/servers/matrix-synapse/0002-Expose-generic-worker-as-binary-under-NixOS.patch:4]: Let's see: https://github.com/matrix-org/synapse/pull/11449

**Pull request merged**